### PR TITLE
Adding more restrictions

### PR DESCRIPTION
**Pull request by lawzlo**
Change Adult Restriction Filtering mechanism to filter all Uris with a restriction as well as unscrapable ones

@stkem @leogrim @yingjieMiao

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/model/NormalizedURIRepo.scala [hunk 1]
```diff
@@ -199,12 +200,16 @@ class NormalizedURIRepoImpl @Inject() (
     { for (r <- rows if r.restriction === targetRestriction) yield r }.list
   }
 
-  def getRestrictionStatusOfURIs(targetRestriction: Restriction, uriIds: Seq[Id[NormalizedURI]])(implicit session: RSession): Seq[Boolean] = {
+  def checkUnrestrictedURIs(uriIds: Seq[Id[NormalizedURI]])(implicit session: RSession): Seq[Boolean] = {
     { for (r <- rows if r.id.inSet(uriIds)) yield r.restriction }.list.map { restriction =>
-      restriction.map(_ == targetRestriction).getOrElse(false)
+      restriction.map(_ => false).getOrElse(true)
```
Comment (stkem) [inline kifi-backend/modules/shoebox/app/com/keepit/model/NormalizedURIRepo.scala]: `restiction.isEmpty` does what you want, I think

Comment (stkem) [inline kifi-backend/modules/shoebox/app/com/keepit/model/NormalizedURIRepo.scala]: actually, you might be able to do `r.restriction.isNull` in the query to get the same thing. Not 100% sure that will compile though.

**Comment by stkem** [comment]
`getAdultRestrictionOfURIs` should be renamed since it's semantics changed. Maybe `getUnrestrictedStatusOfURIs` or `getRecommendableStatusOfURIs` to something like that?
Other than that, I made small optimization comment regarding the db lookup. lgtm otherwise.

**Comment by lawzlo** [comment]
I think I've changed to `checkUnrestrictedURIs`, is that what we want?

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/NormalizedUriCommander.scala [hunk 2]
```diff
@@ -12,9 +12,11 @@ class NormalizedURICommander @Inject() (
     normalizedURIRepo: NormalizedURIRepo,
     db: Database) extends Logging {
 
-  def getAdultRestrictionOfURIs(uris: Seq[Id[NormalizedURI]]): Future[Seq[Boolean]] = {
+  def getCandidateURIs(uris: Seq[Id[NormalizedURI]]): Future[Seq[Boolean]] = {
     db.readOnlyReplicaAsync { implicit session =>
-      normalizedURIRepo.getRestrictionStatusOfURIs(Restriction.ADULT, uris)
+      val adultUris = normalizedURIRepo.checkUnrestrictedURIs(uris)
```
Comment (lawzlo) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/NormalizedUriCommander.scala]: I will rename val name here to unrestrictedURIs

**Comment by stkem** [comment]
It looks like at least the method in the service client still has the old name.

**Comment by lawzlo** [comment]
Oh. I see. I will change it. Thanks.

**Comment by lawzlo** [comment]
Tested. Pushing to Prod.

**Pull request merged**